src/common/models/app-settings/app-settings.ts

```diff
@@ -15,247 +15,167 @@
-    this.version = version || 0;
-    this.clusters = clusters;
-    this.customization = customization;
-    this.dataCubes = dataCubes;
-  }
-
-  public valueOf(): AppSettingsValue {
-    return {
-      version: this.version,
-      clientTimeout: this.clientTimeout,
-      clusters: this.clusters,
-      customization: this.customization,
-      dataCubes: this.dataCubes
-    };
-  }
-
-  public toJS(): AppSettingsJS {
-    let js: AppSettingsJS = {};
-    if (this.version) js.version = this.version;
-    js.clientTimeout = this.clientTimeout;
-    js.clusters = this.clusters.map(cluster => cluster.toJS());
-    js.customization = this.customization.toJS();
-    js.dataCubes = this.dataCubes.map(dataCube => dataCube.toJS());
-    return js;
-  }
-
-  public toJSON(): AppSettingsJS {
-    return this.toJS();
-  }
-
-  public toString(): string {
-    return `[AppSettings v${this.version} dataCubes=${this.dataCubes.length}]`;
-  }
-
-  public equals(other: AppSettings): boolean {
-    return AppSettings.isAppSettings(other) &&
-      this.version === other.version &&
-      this.clientTimeout === other.clientTimeout &&
-      immutableArraysEqual(this.clusters, other.clusters) &&
-      immutableEqual(this.customization, other.customization) &&
-      immutableArraysEqual(this.dataCubes, other.dataCubes);
-  }
-
-  public toClientSettings(): AppSettings {
-    let value = this.valueOf();
-
-    value.clusters = value.clusters.map(c => c.toClientCluster());
-
-    value.dataCubes = value.dataCubes
-      .filter(ds => ds.isQueryable())
-      .map(ds => ds.toClientDataCube());
-
-    return new AppSettings(value);
-  }
-
-  public getVersion(): number {
-    return this.version;
-  }
-
-  public getDataCubesForCluster(clusterName: string): DataCube[] {
-    return this.dataCubes.filter(dataCube => dataCube.clusterName === clusterName);
-  }
-
-  public getDataCube(dataCubeName: string): DataCube {
-    return NamedArray.findByName(this.dataCubes, dataCubeName);
-  }
-
-  public addOrUpdateDataCube(dataCube: DataCube): AppSettings {
-    let value = this.valueOf();
-    value.dataCubes = NamedArray.overrideByName(value.dataCubes, dataCube);
-    return new AppSettings(value);
-  }
-
-  public deleteDataCube(dataCube: DataCube): AppSettings {
-    let value = this.valueOf();
-    const index = value.dataCubes.indexOf(dataCube);
-
-    if (index === -1) {
-      throw new Error(`Unknown dataCube : ${dataCube.toString()}`);
-    }
-
-    let newDataCubes = value.dataCubes.concat();
-    newDataCubes.splice(index, 1);
+export interface SerializedClientAppSettings {
+  version: number;
+  clientTimeout: number;
+  clusters: ClusterJS[];
+  customization: CustomizationJS;
+  dataCubes: DataCubeJS[];
+}
 
-    value.dataCubes = newDataCubes;
-    return new AppSettings(value);
-  }
+export interface ClientAppSettings {
+  version: number;
+  clientTimeout: number;
+  clusters: Cluster[];
```

Comment:
Do we need clusters on the client?

Reply:
Probably not.

Reply:
Right now we need, because DataCube constructor needs it. As with everything "immutable", constructor mus handle cases on front- and back-end :/ If we make `DataCube` classes for server and client separate we could remove clusters from client.